### DISCUSSION TITLE
Can SLEAP keep track of optogenetic light stimulation with laser

### DISCUSSION THREAD

**Discussion by carolinafsoares** [talmolab/sleap#1167] [Help!]
Hi! Can SLEAP keep track of when light from a laser turns on for some optogenetic experiments? Would labelling the light source work? Can it account and keep track of the seconds the light is on vs off. (the light difference is visible in the videos I'm trying to upload).

**Comment by roomrys** [comment]
Hi @carolinafsoares,

This is a novel application, but it should be possible! If we can see it, then SLEAP should be able to track it.

I would suggest starting by labeling just a few frames 5-10, then doing a round of training to check the performance. As for keeping account of the timing, this will need to be calculated during analysis using the frames per second and number of frames that the light was on.

Let us know how it goes and/or if you need any help!

Thanks,
Liezl

**Reply by carolinafsoares** [reply]
Hi @roomrys  Thank you for your reply!

Should I simply create a node and mark it as visible when the light is on and mark it as not visible when the light is off?

Best wishes,
Carolina

**Reply by roomrys** [reply]
Hi @carolinafsoares,

Sorry for the confusion, but by "application" I meant that I have not heard of anyone using SLEAP for the "implementation" you mentioned (tracking the light on/off) - so it would be an experiment, that I suspect will work.

If you are using a single node skeleton for just the light, then I would only label the light when it is turned on (and skip any labeling when the light is off). It has come to our attention that the bottom-up model will not work for single node skeletons since we expect at least one edge in the skeleton for Part Affinity Fields.

If you are using a multi-node skeleton where the light is just one node of many, then I would labeled the node as visible when it is on and non-visible when it is off.

As always, do a test by first labeling a few frames 10-15 and then we can mess with any training parameters if needed.

Thanks,
Liezl
